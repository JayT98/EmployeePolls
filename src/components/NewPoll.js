import {connect} from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useState } from 'react';
import { handleAddQuestion } from '../actions/questions';

const NewPoll = ({dispatch}) => {
    const navigate = useNavigate();
    const [firstOption, setFirstOption] = useState('');
    const [secondOption, setSecondOption] = useState('');

    const handleFirstOptionChange = (e) => {
        const value = e.target.value;
        setFirstOption(value);
    }

    const handleSecondOptionChange = (e) => {
        const value = e.target.value;
        setSecondOption(value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(handleAddQuestion(firstOption,secondOption));
        navigate('/');
    }

    return(
        <div>
            <h1 className='text-3xl font-bold mt-9 text-center'>Would You Rather</h1>
            <p className='text-center text-slate-700/70'>Create Your Own Poll</p>

            <form onSubmit={handleSubmit}>
                <div className='mt-9'>
                    <h3 htmlFor='firstOption'
                            data-testid='firstOptionLabel'
                            className='block text-sm font-bold text-black text-center'>
                        First Option
                    </h3>
                    <div className=' mt-3'>
                        <input value={firstOption}
                                onChange={handleFirstOptionChange}
                                type='text'
                                name='firstOption'
                                id='firstOption'
                                placeholder='Option One'
                                data-testid='firstOption'
                                className='px-3 py-2 bg-white border shadow-sm border-slate-300 placeholder-slate-400  disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-pink-500 invalid:text-pink-600 focus:invalid:border-pink-500 focus:invalid:ring-pink-500 disabled:shadow-none'></input>
                    </div>
                </div>

                <div className='mt-9'>
                    <h3 htmlFor='secondOption'
                            data-testid='secondOptionLabel'
                            className='block text-sm font-bold text-black text-center'>
                        Second Option
                    </h3>
                    <div className=' mt-3'>
                        <input value={secondOption}
                                onChange={handleSecondOptionChange}
                                type='text'
                                name='secondOption'
                                id='secondOption'
                                placeholder='Option Two'
                                data-testid='secondOption'
                                className='px-3 py-2 bg-white border shadow-sm border-slate-300 placeholder-slate-400  disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-pink-500 invalid:text-pink-600 focus:invalid:border-pink-500 focus:invalid:ring-pink-500 disabled:shadow-none'></input>
                    </div>
                </div>

                <div className="mt-6 text-center">
                    <button type="submit"
                            data-testid="submit-poll"
                            className="bg-green-500 hover:bg-green-700 px-5 py-2.5 text-sm leading-5 rounded-md font-bold text-white">
                        Submit
                    </button>
                </div>
            </form>
        </div>
    )
}

export default connect()(NewPoll);