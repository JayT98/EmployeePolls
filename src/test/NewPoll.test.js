import {fireEvent, render} from "@testing-library/react";
import {Provider} from "react-redux";
import {store} from "../store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import NewPoll from "../components/NewPoll";

describe("NewPoll", () => {
    it("should render the component", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPoll/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });

    it("should display all elements", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPoll/>
                </BrowserRouter>
            </Provider>
        );

        const firstOptionLabelElement = component.getByTestId("firstOptionLabel");
        const firstOptionInputElement = component.getByTestId("firstOption");
        const secondOptionLabelElement = component.getByTestId("secondOptionLabel");
        const secondOptionInputElement = component.getByTestId("secondOption");
        const submitButtonElement = component.getByTestId("submit-poll");

        expect(firstOptionLabelElement.textContent).toBe("First Option");
        expect(secondOptionLabelElement.textContent).toBe("Second Option");
        expect(submitButtonElement.textContent).toBe("Submit");

        fireEvent.change(firstOptionInputElement, {target: {value: 'Texas'}});
        fireEvent.change(secondOptionInputElement, {target: {value: 'New Hampshire'}});
        expect(firstOptionInputElement.value).toBe("Texas");
        expect(secondOptionInputElement.value).toBe("New Hampshire");
    });
});
