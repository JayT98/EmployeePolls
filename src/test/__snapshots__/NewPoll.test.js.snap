// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="text-3xl font-bold mt-9 text-center"
        >
          Would You Rather
        </h1>
        <p
          class="text-center text-slate-700/70"
        >
          Create Your Own Poll
        </p>
        <form>
          <div
            class="mt-9"
          >
            <h3
              class="block text-sm font-bold text-black text-center"
              data-testid="firstOptionLabel"
              for="firstOption"
            >
              First Option
            </h3>
            <div
              class=" mt-3"
            >
              <input
                class="px-3 py-2 bg-white border shadow-sm border-slate-300 placeholder-slate-400  disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-pink-500 invalid:text-pink-600 focus:invalid:border-pink-500 focus:invalid:ring-pink-500 disabled:shadow-none"
                data-testid="firstOption"
                id="firstOption"
                name="firstOption"
                placeholder="Option One"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="mt-9"
          >
            <h3
              class="block text-sm font-bold text-black text-center"
              data-testid="secondOptionLabel"
              for="secondOption"
            >
              Second Option
            </h3>
            <div
              class=" mt-3"
            >
              <input
                class="px-3 py-2 bg-white border shadow-sm border-slate-300 placeholder-slate-400  disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-pink-500 invalid:text-pink-600 focus:invalid:border-pink-500 focus:invalid:ring-pink-500 disabled:shadow-none"
                data-testid="secondOption"
                id="secondOption"
                name="secondOption"
                placeholder="Option Two"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="mt-6 text-center"
          >
            <button
              class="bg-green-500 hover:bg-green-700 px-5 py-2.5 text-sm leading-5 rounded-md font-bold text-white"
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="text-3xl font-bold mt-9 text-center"
      >
        Would You Rather
      </h1>
      <p
        class="text-center text-slate-700/70"
      >
        Create Your Own Poll
      </p>
      <form>
        <div
          class="mt-9"
        >
          <h3
            class="block text-sm font-bold text-black text-center"
            data-testid="firstOptionLabel"
            for="firstOption"
          >
            First Option
          </h3>
          <div
            class=" mt-3"
          >
            <input
              class="px-3 py-2 bg-white border shadow-sm border-slate-300 placeholder-slate-400  disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-pink-500 invalid:text-pink-600 focus:invalid:border-pink-500 focus:invalid:ring-pink-500 disabled:shadow-none"
              data-testid="firstOption"
              id="firstOption"
              name="firstOption"
              placeholder="Option One"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="mt-9"
        >
          <h3
            class="block text-sm font-bold text-black text-center"
            data-testid="secondOptionLabel"
            for="secondOption"
          >
            Second Option
          </h3>
          <div
            class=" mt-3"
          >
            <input
              class="px-3 py-2 bg-white border shadow-sm border-slate-300 placeholder-slate-400  disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 focus:outline-none focus:border-sky-500 focus:ring-sky-500 block w-full rounded-md sm:text-sm focus:ring-1 invalid:border-pink-500 invalid:text-pink-600 focus:invalid:border-pink-500 focus:invalid:ring-pink-500 disabled:shadow-none"
              data-testid="secondOption"
              id="secondOption"
              name="secondOption"
              placeholder="Option Two"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="mt-6 text-center"
        >
          <button
            class="bg-green-500 hover:bg-green-700 px-5 py-2.5 text-sm leading-5 rounded-md font-bold text-white"
            data-testid="submit-poll"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
